util/flags/tls.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package flags implements the webhook server TLS options utilities.
+package flags
+
+import (
+	"crypto/tls"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/pflag"
+	cliflag "k8s.io/component-base/cli/flag"
+)
+
+// SecureWebhookOptions has the options to configure the TLS settings for a webhook server.
+type SecureWebhookOptions struct {
```

Comment:
nitpicking on naming, feel free to say no:

What about `TLSOptions` (and for consistency `AddTLSOptionsFlags`)

Reply:
I was initially leaning towards `TLSOptions` as well but wanted to express the intent of the struct, hence chose `SecureWebhookOptions`. But I am not too sold on the name, I am going to revert it to `TLSOptions` and with the package name `flags.TLSOptions` it expresses the intent better.